controllers/nicclusterpolicy_controller.go

```diff
@@ -212,24 +225,35 @@ func (r *NicClusterPolicyReconciler) handleMOFEDWaitLabelsNoConfig(ctx context.C
 		} else if node.GetLabels()[nodeinfo.NodeLabelMlnxNIC] == "true" {
 			labelValue = "false"
 		}
-		if node.GetLabels()[nodeinfo.NodeLabelWaitOFED] == labelValue {
-			// already has the right value
-			continue
-		}
 		if err := setOFEDWaitLabel(ctx, r.Client, node.Name, labelValue); err != nil {
```

Comment:
btw should we still attempt to patch node obj if its already patched to the correct value ?
or you want to avoid cache edge cases when node comes back to cluster with/without mlnx nic ?

Reply:
yes, I decided not to rely on cached data for Pods. We don't watch Pods in the API that means that we may have no chance to fix the state on the Pod update event.

Reply:
we always patch the object to avoid possible cache edge cases